src/item.cpp

```diff
@@ -306,12 +248,45 @@ bool item::covers( const body_part bp ) const
         // go on another bodypart.
         return bp == bp_torso;
     }
-    return covered_bodyparts.test( bp );
+    return get_covered_body_parts().test(bp);
 }
 
-const std::bitset<num_bp> &item::get_covered_body_parts() const
+const std::bitset<num_bp> item::get_covered_body_parts() const
 {
-    return covered_bodyparts;
+    auto res = type->armor->covers;
+
+    switch (get_side()) {
+        case LEFT:
+            res.reset(bp_arm_r);
+            res.reset(bp_hand_r);
+            res.reset(bp_leg_r);
+            res.reset(bp_foot_r);
+            break;
+
+        case RIGHT:
+            res.reset(bp_arm_l);
+            res.reset(bp_hand_l);
+            res.reset(bp_leg_l);
+            res.reset(bp_foot_l);
+            break;
+    }
+
+    return res;
+}
+
+bool item::is_sided() const {
+    auto t = find_armor_data();
+    return t ? t->sided : false;
+}
+
+int item::get_side() const {
+    return get_var("lateral", BOTH);
+}
+
+bool item::set_side (side s) {
+    if (!is_sided()) return false;
+    set_var("lateral", s);
```

Comment:
It might be good to erase the variable (`erase_var`) instead of setting it to `BOTH`. Items with different variables don't stack. A default created item doesn't have the variable. An item that is dropped has its "lateral" variable set to "BOTH". Those two items don't stack. Erasing the variable makes the dropped item equal to the default created item and they will stack.